exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/auth/DrillHttpSecurityHandlerProvider.java

```diff
@@ -114,23 +112,27 @@ public DrillHttpSecurityHandlerProvider(DrillConfig config, DrillbitContext dril
   }
 
   @Override
-  public void doStart() throws Exception {
+  protected void doStart() throws Exception {
     super.doStart();
     for (DrillHttpConstraintSecurityHandler securityHandler : securityHandlers.values()) {
       securityHandler.doStart();
     }
   }
 
   @Override
-  public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
-      throws IOException, ServletException {
+  public boolean handle(Request request, Response response, Callback callback) throws Exception {
+    // Get servlet request/response from the core request/response
+    org.eclipse.jetty.ee10.servlet.ServletContextRequest servletContextRequest =
+        org.eclipse.jetty.server.Request.as(request, org.eclipse.jetty.ee10.servlet.ServletContextRequest.class);
+    HttpServletRequest httpServletRequest = servletContextRequest.getServletApiRequest();
+    HttpServletResponse httpServletResponse = servletContextRequest.getHttpServletResponse();
```

Comment:
This class is a `Handler.Wrapper`
It has no access to the `ee10` implementation.
Converting from core `Request` / `Response` to Servlet `HttpServletRequest` and `HttpServletResponse` can only be done within a `ServletContext`, which this handler does not participate in.

If you need this, then extend from the `ConstraintSecurityHandler` from the ee10 layer.

The `org.eclipse.jetty.ee10.servlet.security.ConstraintSecurityHandler` is the appropriate `ee10` specific implementation of `org.eclipse.jetty.security.SecurityHandler` and will be inserted into the `WebAppContext` (or `ServletContextHandler`) in the right place if your class uses the correct type.